packages/runtime-core/src/renderer.ts

```diff
@@ -779,7 +779,12 @@ function baseCreateRenderer(
       invokeDirectiveHook(n2, n1, parentComponent, 'beforeUpdate')
     }
 
-    if (__DEV__ && parentComponent && parentComponent.hmrUpdated) {
+    if (
+      __DEV__ &&
+      parentComponent &&
+      (parentComponent.hmrUpdated ||
```

Comment:
In the case of mutliple nested HOCs, we should probably walk and exhaust the parent chain to check for hmr updates.

Reply:
Actually this fix wasn't actually enough, after implementing a test with nested HOC, found some other issues 👍 

The added test might be a bit too much 😄